storage-proofs/src/drgporep.rs

```diff
@@ -327,7 +327,7 @@ where
                 let extracted = decode_domain_block::<H>(
                     pub_params.sloth_iter,
                     &pub_inputs.replica_id.expect("missing replica_id"),
-                    tree_r.as_ref(),
+                    tree_r,
```

Comment:
@schomatis thinking about this some more, this method only needs the leaves, which we could still provide as slice, avoiding the need for the create_key_from_tree method below,

Reply:
@dignifiedquire With the new offload logic I can't expose any internals of the `mmap` stores (which I see as a plus) because I need to control when they are valid and when they're not, can't let you hold a slice to something that isn't in memory. At least I didn't see a way around it without breaking too much stuff, were you thinking about a specific implementation?

Reply:
Slowly starting to get worried about the implications of not being able to read slices anymore. 

What we could do that would solve that problem I believe is to provide the following method 

```rust
fn read_into(&self, &mut [u8]) {}
```

which would read into a preallocated slice, instead of allocating a new vec everytime we need to read sth. This could be then used in the `create_key_from_tree` to read into the hasher directly instead of all the allocations 

--- 

Independent of the performance implications I guess we have to have two different methods for now :(

Reply:
Sounds good, I'll implement that on the `merkle_light` side.